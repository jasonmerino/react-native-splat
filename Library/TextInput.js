import React, {
  Component,
  PropTypes,
} from 'react';
import ReactNative from 'react-native';
console.log('MEH', ReactNative.StyleSheet);

const componentStyles = ReactNative.StyleSheet.create({
  input: {
    height: 48,
    padding: 10,
    borderRadius: 3,
    backgroundColor: '#fff',
  },
});

class TextInput extends Component {
  getStyle() {
    return [this.props.style, componentStyles.input];
  }

  render() {
    const {
      onChangeText,
      placeholder,
      keyboardType,
      value,
      secureTextEntry,
    } = this.props;
    return (
      <ReactNative.TextInput
        placeholder={placeholder}
        value={value}
        keyboardType={keyboardType}
        onChangeText={onChangeText}
        secureTextEntry={secureTextEntry}
        style={this.getStyle()}
      />
    );
  }
}

TextInput.propTypes = {
  onChangeText: PropTypes.func,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  keyboardType: PropTypes.oneOf([
    'default',
    'email-address',
    'numeric',
    'phone-pad',
    'ascii-capable',
    'numbers-and-punctuation',
    'url',
    'number-pad',
    'name-phone-pad',
    'decimal-pad',
    'twitter',
    'web-search',
  ]),
  secureTextEntry: PropTypes.bool,
  style: ReactNative.Text.propTypes.style,
};

TextInput.defaultProps = {
  secureTextEntry: false,
  placeholder: '',
  keyboardType: 'default',
  style: {},
};

export default TextInput;
