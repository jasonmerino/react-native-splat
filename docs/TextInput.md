`TextInput` (component)
=======================



Props
-----

### `style`

type: `custom`
defaultValue: `{}`


### `underlineColorAndroid`

defaultValue: `'transparent'`

