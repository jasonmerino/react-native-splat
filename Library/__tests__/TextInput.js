import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from '../TextInput';


it('renders correctly', () => {
  const tree = renderer.create(
    <TextInput />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
