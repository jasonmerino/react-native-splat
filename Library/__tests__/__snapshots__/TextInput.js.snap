exports[`test renders correctly 1`] = `
<TextInput
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 3,
        "height": 48,
        "padding": 10,
      },
    ]
  }
  underlineColorAndroid="transparent" />
`;
